yarn/history/src/main/scala/org/apache/spark/deploy/history/yarn/YarnHistoryService.scala

```diff
@@ -0,0 +1,1048 @@
+    if (!aclsOn) {
+      logDebug("ACLs are disabled; not creating the timeline domain")
+      return None
+    }
+    val predefDomain = sparkConf.getOption(TIMELINE_DOMAIN)
+    if (predefDomain.isDefined) {
+      logDebug(s"Using predefined domain $predefDomain")
+      return predefDomain
+    }
+    val current = UserGroupInformation.getCurrentUser.getShortUserName
+    val adminAcls  = stringToSet(sparkConf.get("spark.admin.acls", ""))
+    val viewAcls = stringToSet(sparkConf.get("spark.ui.view.acls", ""))
+    val modifyAcls = stringToSet(sparkConf.get("spark.modify.acls", ""))
+
+    val readers = (Seq(current) ++ adminAcls ++ modifyAcls ++ viewAcls).mkString(" ")
+    val writers = (Seq(current) ++ adminAcls ++ modifyAcls).mkString(" ")
+    var domain = DOMAIN_ID_PREFIX + _applicationId
+    logInfo(s"Creating domain $domain with readers: $readers and writers: $writers")
+
+    // create the timeline domain with the reader and writer permissions
+    val timelineDomain = new TimelineDomain()
+    timelineDomain.setId(domain)
+    timelineDomain.setReaders(readers)
+    timelineDomain.setWriters(writers)
+    try {
+      timelineClient.putDomain(timelineDomain)
+      Some(domain)
+    } catch {
+      case e: Exception => {
+        logError(s"cannot create the domain $domain", e)
+        // fallback to default
+        None
+      }
+    }
+  }
+
+  /**
+   * Start the service, calling the service's `init()` and `start()` actions in the
+   * correct order
+   * @param binding binding to the spark application and YARN
+   */
+  override def start(binding: YarnExtensionServiceBinding): Unit = {
+    val oldstate = enterState(StartedState)
+    if (oldstate != CreatedState) {
+      // state model violation
+      _serviceState.set(oldstate)
+      throw new IllegalArgumentException(s"Cannot start the service from state $oldstate")
+    }
+    val context = binding.sparkContext
+    val appId = binding.applicationId
+    val attemptId = binding.attemptId
+    require(context != null, "Null context parameter")
+    bindToYarnApplication(appId, attemptId)
+    this.sparkContext = context
+
+    this.config = new YarnConfiguration(context.hadoopConfiguration)
+
+    val sparkConf = sparkContext.conf
+
+    // work out the attempt ID from the YARN attempt ID. No attempt, assume "1".
+    // this is assumed by the AM, which uses it when creating a path to an attempt
+    val attempt1 = attemptId match {
+      case Some(attempt) => attempt.getAttemptId.toString
+      case None => CLIENT_BACKEND_ATTEMPT_ID
+    }
+    setContextAppAndAttemptInfo(Some(appId.toString), Some(attempt1))
+    _batchSize = sparkConf.getInt(BATCH_SIZE, _batchSize)
+    _postQueueLimit = sparkConf.getInt(POST_QUEUE_LIMIT, _postQueueLimit)
+    retryInterval = 1000 * sparkConf.getTimeAsSeconds(POST_RETRY_INTERVAL,
+      DEFAULT_POST_RETRY_INTERVAL)
+    shutdownWaitTime = 1000 * sparkConf.getTimeAsSeconds(SHUTDOWN_WAIT_TIME,
```

Comment:
see above. I just went with the same default convention as that for the FS history provider. Happy to change it though.